### PR TITLE
feat(table): implement basic view for dot table component

### LINKED ISSUE #27939: Implement Basic View for Dot Table Component
- [ ] Add sortable columns functionality.
- [ ]  Apply mouse-over row hover color.

### PR DESCRIPTION
**Pull request by manuelrojas**
### Proposed Changes
* Adding the basic table component

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
image  | 
 

https://github.com/dotCMS/core/assets/3438705/2f0f24c2-1f41-4ee4-b451-e9148634c1da

### CODE REVIEW

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        border-radius: $border-radius-md;
+    }
+
+    .p-datatable .p-datatable-thead > tr > th {
+        background-color: $color-palette-gray-200;
+        color: $black;
+    }
+
+    .p-datatable .p-datatable-tbody > tr:hover {
+        background-color: #ecf0fd;
```
Comment (zJaaal) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss:18]: Should we have a variable?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        border-radius: $border-radius-md;
+    }
+
+    .p-datatable > .p-datatable-wrapper {
+        border-radius: $border-radius-md;
+    }
+
+    .p-datatable .p-datatable-thead > tr > th {
+        background-color: $color-palette-gray-200;
+        color: $black;
```
Comment (zJaaal) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss:14]: should we use nested rules instead of `>`?

what do you think? @fmontes

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss (hunk 2 above)
Comment (fmontes) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss:14]: Good question... I'll guess it depends of what you are doing here.

Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss:14]: Take advantage of the nested scss feature.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+// Function to add a day to a date
+const addDays = (date, days) => {
+    const result = new Date(date);
+    result.setDate(result.getDate() + days);
+
+    return result;
+};
+
+interface File {
```
Comment (jdotcms) [inline core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts:20]: a File usually is a name or path and some meta info.
This definition looks a bit more stuff, probably a diff name would be good

File: core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        docs: {
+            description: {
+                component:
+                    'Table is a container component to display data in tabular format.: https://primeng.org/table'
+            }
+        }
+    }
+} as Meta;
+
+export const Default: Story = () => ({
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts:118]: use `StoryFn` instead

File: core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts (hunk 3 above)
Comment (KevinDavilaDotCMS) [reply core-web/apps/dotcms-ui/src/stories/primeng/table/Table.stories.ts:20]: Must to rename this interface

`File` is a existing interface inside Typescript
When we create a interface that alredy exist, both interfaces are merged
So this File interface have all attributes of File (binarys) like `lastModified, code, buffer, etc`

**Review by KevinDavilaDotCMS — CHANGES_REQUESTED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**